perf/exporters.go

```diff
@@ -57,7 +57,7 @@ func NewCloudExporter(opts CloudOptions) (Exporter, error) {
 		zone := os.Getenv("GCP_COMPUTE_ZONE")
 		if zone == "" {
 			// Get from GCP Metadata
-			if zone, err = metadata.Zone(); err != nil {
+			if zone, err = metadata.ZoneWithContext(sdOpts.Context); err != nil {
```

Comment:
could see just using a background context as well, not clear if the context used here is the right one